### DISCUSSION TITLE
[✨] useRouteSegments

### DISCUSSION THREAD

**Discussion by brandonpittman** [QwikDev/qwik-evolution#143] [Proposals For Qwik]
### Is your feature request related to a problem?

I'd like to be able to access information about each route segment. I'm trying to recreate something similar to Remix's [`useMatches`](http://remix.run/docs/en/main/hooks/use-matches). It's handy to set some data in a route or layout and be able to fetch it in a parent layout. Think breadcrumbs.

### Describe the solution you'd like

I'd like to be able to call `useRouteSegements` or `useMatches` and get back an array with each matched route segment's `head` export. Remix has a `handle` export they use for this, but I think we could reuse the `head` export and stick stuff in the `frontmatter` key.

### Describe alternatives you've considered

I've been playing around with adding stuff to a layout or route's `frontmatter` in the `head` export. It kind of works, but you need each declaration to have some knowledge of its possible child routes. If we had an array with each matched route segment, we could just declare this kind of data and read it in a parent layout:

```ts
return {
  frontmatter: {
    breadcrumb: "Some Child Route"
  }
}
```

Then, hopefully, the `useRouteSegments` function would return something like this:

```ts
type RouteSegment = { 
  head: DocumentHead; 
  path: string; // the path matching the layout/route
}

// return value
RouteSegment[]
```

**Comment by gioboa** [comment]
We moved this issue to `qwik-evolution` repo to create a RFC discussion for this.
Here is our [Qwik RFC process](https://github.com/QwikDev/qwik-evolution?tab=readme-ov-file#qwik-rfc-process) thanks.
